common/src/main/java/com/microsoft/identity/common/internal/util/ParcelableUtil.java

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.util;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+import androidx.annotation.NonNull;
+
+public class ParcelableUtil {
```

Comment:
Should we have documentation on this public class?  Should it be final, with a private constructor?  Higher level, is there a reason to prefer static method references over instance methods on a singleton-safe class?

Reply:
Added documentation via https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/918/commits/477877f2d99d6eb7d752f18d941a682b51665c62. The intention of this class is have just `util` methods to transform parcelable , hence static methods to keep it simple. I would say it's a coding preference we have been following for all our `Util` classes.

Reply:
I'm fine with this, though it will make it more difficult to induce exceptional behavior in the writeToParcel methods for the purposes of testing.  What I would suggest is that, since it's never reasonable to instantiate or inherit from this type of class, declare it to be final and give it a private constructor.

Reply:
yup, I would not expect to induce any behavior to these as these are basic util methods using Platform API's to transform. Declared as `final` and added a `private` constructor as suggested

Reply:
Thanks!   I'm looking into static analysis to enforce this kind of thing so we're not doing it in review.  Looks like we have a checkstyle ruleset in the codebase...